public/app/features/explore/extensions/getExploreExtensionConfigs.tsx

```diff
@@ -38,6 +41,21 @@ export function getExploreExtensionConfigs(): PluginExtensionLinkConfig[] {
           });
         },
       }),
+      createExtensionLinkConfig<PluginExtensionExploreContext>({
+        title: 'Add Correlation',
+        description: 'Create a correlation from this query',
+        extensionPointId: PluginExtensionPoints.ExploreToolbarAction,
+        icon: 'link',
+        configure: (context) => {
+          return context?.shouldShowAddCorrelation ? {} : undefined;
+        },
+        onClick: (_, { context }) => {
+          dispatch(changeCorrelationEditorDetails({ editorMode: true }));
+          if (context?.exploreId) {
```

Comment:
In what case `exploreId` is not defined?

Reply:
I think I just reflexively did this for the typescript error, but youre right, theres no way you could click this button without an explore id existing. Fixed.